herddb-core/src/main/java/herddb/data/integrity/TableDataChecksum.java

```diff
@@ -0,0 +1,91 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package herddb.data.integrity;
+
+import herddb.model.DataScanner;
+import herddb.model.DataScannerException;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import herddb.model.TransactionContext;
+import herddb.model.FullTableScanPredicate;
+import herddb.model.StatementEvaluationContext;
+import herddb.model.commands.ScanStatement;
+import herddb.core.TableSpaceManager;
+import herddb.utils.DataAccessor;
+import herddb.codec.RecordSerializer;
+import herddb.model.Column;
+import net.jpountz.xxhash.StreamingXXHash64;
+import net.jpountz.xxhash.XXHashFactory;
+/**
+ *
+ * @author Hamado.Dene
+ */
+public class TableDataChecksum{
```

Comment:
you can make this class 'abstract' and add a "private" costructor (search for "Utility class")